### PR TITLE
Add Standard Toggles

### PR DESCRIPTION
**Pull request by phariel**
##### Action item:
1. I'll write detail descriptions later on component-ef.
2. Continue to refine event binding (done).
3. Add $.fn & noConflict support (done).
4. Rebase for squashing commits.

### CODE REVIEW

**Comment by phariel** [comment]
@mikaelkaron @normanzb @faceach @jacksky007 would you please review & someone merge this, thanks. :monkey_face:

File: dist/js/bootstrap.js [hunk 1]
```diff
@@ -1988,6 +1988,224 @@ if (!jQuery) { throw new Error("Bootstrap requires jQuery") }
...
+      var $btn = $this.find('.btn')
+      var data = $this.data(toggleBase)
+      var val = data === true
+
+      $this.on(toggleFocus, function(e) {
+        if ($this.is('.disabled, [disabled]')) return
+        // TODO: make clear that bind which key to trigger toggle when press 'tab' key and focus on toggle
+        //       Current are: Enter & Space
+        $this.on(toggleKeydown, function(e) {
+          if (/(13|32)/.test(e.keyCode)) {
```
Comment (mikaelkaron) [inline dist/js/bootstrap.js]: is `keyCode` a `Number`? In that case I'd rather us do `if (e.keyCode === 13 || e.keyCode === 32)` here

Comment (phariel) [inline dist/js/bootstrap.js]: @mikaelkaron yes but when you look at [dropdown.js](https://github.com/englishtown/bootstrap/blob/develop/js/dropdown.js) line 63, you could find this format.
I just refer to this and want our coding style consistent. :smile_cat:

**Comment by phariel** [comment]
@mikaelkaron @normanzb @faceach @jacksky007 merge this or not both okay, I'll change code structure and refine event binding later. This's just a preview, thanks.

**Comment by mikaelkaron** [comment]
I think we  should wait until the whole task is "finished" (meaning not wip anymore) before we merge, but so far I'm ok with it.

@englishtown/frontend The code here matches the other code in bootstrap, but not the conventions we use at englishtown. I personally feel that we should stick with whatever the project is using (so in this case the bootstrap standard). Are we all fine with that?

**Comment by mikaelkaron** [comment]
@phariel btw, you'll probably have to rebase the pr before it can be applied, but you can do that in the end (and at that time we should probably squash the commits to)

**Comment by jacksky007** [comment]
Frankly speaking, bootstrap's code style on Javascript is partly OK for me, while some parts I think bad for me.

No semicolon if unnecessary is OK for me.
No tab is first principle while coding in all languages.

I dislike declaring multiple variables using only one 'var' keyword.
I also dislike 'comma first'.
I dislike no brace for single line either because we maybe change it to multiple lines in the future.

Here are my opinions, and I'm flexible. If no one of you dislike the points I mentioned as above, I can follow all these rules. That's not very critical problem for me.

**Comment by phariel** [comment]
Standard toggle refined, I'll rebase commits later

**Comment by mikaelkaron** [comment]
There are _tons_ of things I don't like about the coding standards in bootstrap, but I'd rather us follow it anyways simply because it's easier when it all looks the same rather than like 'EF-like'.

So I guess that if we all agree we just say 'follow the bootstrap standards'. Looking at [`CONTRIBUTING.md`](../blob/develop/CONTRIBUTING.md) I see this:

> - No semicolons
> - Comma first
> - 2 spaces (no tabs)
> - strict mode
> - "Attractive"

What's missing is a [`.jshintrc`](http://jshint.com/docs/config/) file that enforces this. Maybe someone feels up to the task and we can send upstream?

**Comment by mikaelkaron** [comment]
Oh, forgot to mention, if you want this to be easier to merge, just exclude the files from `dist` as they are quite unpredictable (and rebuilt all the time anyways)

**Comment by yaoazhen** [comment]
@mikaelkaron 
some update from jshint : http://www.jshint.com/blog/2013-08-02/npm/

**Comment by phariel** [comment]
@mikaelkaron Should we add `dist` folder to `.gitignore` file like school's code? I thought we must run `grunt` first after every pulling and code changing is a good way to get the newest dist files.

**Comment by phariel** [comment]
@mikaelkaron sorry, I forgot that gh-pages must use dist files, please forget what I said.

**Comment by mikaelkaron** [comment]
That's cool. Is this what bootstrap is doing at the moment? I guess that the one thing I'd wonder about there is IDE integration that can read '.jshintrc'

**Comment by phariel** [comment]
Code refined with suggestion of @faceach , thanks buddy. :smiley_cat:

**Comment by phariel** [comment]
@mikaelkaron @faceach @jacksky007 @normanzb guys, if you reviewed and don't have question, please inform me. I'll create a new branch which must be squashed commits for merging, thanks.

**Comment by mikaelkaron** [comment]
You can just squash in this branch! I'm quite OK with the code, but could we add tests? (to mirror the ones for the original components) and drop the files from `dist` (they will always prevent you from doing a clean merge)

**Comment by mikaelkaron** [comment]
Also, sorry we've been slow at merging this, I really appreciate the work put in here and I'm looking forward to having this merged soon!

**Comment by phariel** [comment]
@mikaelkaron something went wrong from beginning, i tried `rebase` and had never succeed. 
I used `git rebase -i` for squashing commits, but there were so many conflicts on `dist` files. 
When I resolved conflicts and `git rebase --continue`, a new `dist` files conflict popped up.
Again and again, drive me crazy.
Could I create a new branch and copy current codes from this branch to it and commit?
This's the direct path to resolve squashing commits issue I think. :crying_cat_face: 

And I would never commit `dist` files again.

**Comment by mikaelkaron** [comment]
Sure. just make a new PR

**Pull request closed without merging**